source/extensions/filters/network/postgres_proxy/postgres_decoder.cc

```diff
@@ -259,16 +255,25 @@ bool DecoderImpl::onData(Buffer::Instance& data, bool frontend) {
     }
   }
 
-  std::vector<MsgAction>& actions = std::get<2>(msg.get());
-  for (const auto& action : actions) {
-    action(this);
-  }
-
   // message_len_ specifies total message length including 4 bytes long
   // "length" field. The length of message body is total length minus size
   // of "length" field (4 bytes).
   uint32_t bytes_to_read = message_len_ - 4;
 
+  std::vector<MsgAction>& actions = std::get<2>(msg.get());
+  if (!actions.empty()) {
```

Comment:
@cpakulski sorry, do we have tests to check this condition? To make sure, if `actions.empty()` we don't call `linearize` at all.

Reply:
Added unit test to verify that.